daemon/Jobs/UpdaterJob.cs

```diff
@@ -195,49 +288,236 @@ public void Perform()
                 // Extract
                 _logger.LogInformation("UJ: Extracting {0} to {1}", targetArchive, targetDirectory);
                 ZipFile.ExtractToDirectory(targetArchive, targetDirectory);
+                _logger.LogDebug("UJ: Archive has been extracted successfully.");
 
                 // Delete the archive after extraction.
                 File.Delete(targetArchive);
+                _logger.LogDebug("UJ: Downloaded version archive has been deleted.");
 
                 // Get the expected symlink path.
                 var latestPath = Path.Combine(RootInstallationDirectory, "latest");
+                _logger.LogDebug("UJ: Latest symlink path has been generated.");
 
                 // Copy the databases
+                _logger.LogDebug("UJ: Getting database migration information.");
+                var databasePaths = GetDatabasePaths();
+                _logger.LogDebug("UJ: {0} database(s) need to be migrated.", databasePaths.Length);
                 foreach (var databasePath in GetDatabasePaths())
                 {
-                    var basename = new FileInfo(databasePath).Name;
-                    var databaseDestPath = Path.Combine(targetDirectory, "daemon", "Database", basename);
-                    File.Copy(databasePath, databaseDestPath);
-                    _logger.LogInformation("UJ: Migrated Database: {0} => {1}", databasePath, databaseDestPath);
+                    try
+                    {
+                        var basename = new FileInfo(databasePath).Name;
+                        var databaseDestinationPath = Path.Combine(targetDirectory, "daemon", "Database", basename);
+                        _logger.LogDebug("UJ: Attempting to copy database {0} to path {1}", databasePath,
+                            databaseDestinationPath);
+                        File.Copy(databasePath, databaseDestinationPath);
+                        _logger.LogInformation("UJ: Migrated Database: {0} => {1}", databasePath,
+                            databaseDestinationPath);
+                    }
+                    catch (Exception exception)
+                    {
+                        AppConfig.UpdateDeadlock = false;
```

Comment:
Instead of a true/false way like this, I would instead recommend a long that holds a unix timestamp that you can perform diffs against.

I'm also not very sure if AppConfig is the right place to hold such a variable. You can have your own UpdateConfig static class if you wish.

An update that's been stuck for more than an hour likely has already gone wrong without the user being none the wiser, etc.

Reply:
I needed something explicitly to check if a previous updater job was running or failed, and didn't finish gracefully. I'll move this out of the application config